src/olympia/lib/crypto/tests/test_signing.py

```diff
@@ -451,213 +451,188 @@ def test_sign_full(self, mock_sign_file):
+            assert self.version.version_int == version_int('0.0.1')
             tasks.sign_addons([self.addon.pk])
             assert mock_sign_file.called
             self.version.reload()
-            assert self.version.version == '1.3'
-            assert self.version.version_int == version_int('1.3')
+            assert self.version.version == '0.0.1'
+            assert self.version.version_int == version_int('0.0.1')
             assert file_hash == self.file_.generate_hash()
             self.assert_no_backup()
 
     @mock.patch('olympia.lib.crypto.tasks.sign_file')
     def test_dont_bump_not_signed(self, mock_sign_file):
         mock_sign_file.return_value = None  # Pretend we didn't sign.
-        fpath = 'src/olympia/files/fixtures/files/jetpack.xpi'
+        fpath = 'src/olympia/files/fixtures/files/webextension.xpi'
         with amo.tests.copy_file(fpath, self.file_.file_path):
             file_hash = self.file_.generate_hash()
-            assert self.version.version == '1.3'
-            assert self.version.version_int == version_int('1.3')
+            assert self.version.version == '0.0.1'
+            assert self.version.version_int == version_int('0.0.1')
             tasks.sign_addons([self.addon.pk])
             assert mock_sign_file.called
             self.version.reload()
-            assert self.version.version == '1.3'
-            assert self.version.version_int == version_int('1.3')
+            assert self.version.version == '0.0.1'
+            assert self.version.version_int == version_int('0.0.1')
             assert file_hash == self.file_.generate_hash()
             self.assert_no_backup()
 
     @mock.patch('olympia.lib.crypto.tasks.sign_file')
-    def test_resign_bump_version_in_model_if_force(self, mock_sign_file):
+    def test_resign_only_current_versions(self, mock_sign_file):
+        fname = './src/olympia/files/fixtures/files/webextension.xpi'
+
+        new_current_version = amo.tests.version_factory(
+            addon=self.addon, version='0.0.2')
+        new_file = new_current_version.current_file
+
+        with amo.tests.copy_file(fname, new_file.file_path):
+            file_hash = new_file.generate_hash()
+            tasks.sign_addons([self.addon.pk])
+
+            # Only one signing call since we only sign the most recent
+            # versions
+            assert mock_sign_file.call_count == 1
+
+            new_current_version.reload()
+            assert new_current_version.version == '0.0.2.1-signed'
+            assert new_current_version.version_int == version_int(
+                '0.0.2.1-signed')
+            assert file_hash != new_file.generate_hash()
+
+            # Verify that the old version hasn't been resigned
+            self.version.reload()
+            assert self.version.version == '0.0.1'
+            assert self.version.version_int == version_int('0.0.1')
+            assert file_hash != new_file.generate_hash()
```

Comment:
This doesn't make sense, it has nothing to do with `self.version`. Wrong copy/paste ?

Reply:
Fixed in 63f046794a, indeed, that didn't make any sense :roll_eyes: yeah, some copy&paste error :-/